### PR TITLE
Manual APK release by flavor workflow configuration

### LINKED ISSUE #2883: APK Release Automation for Multiple Flavors
**Describe the Issue**
We need to improve the release management process for APKs via CI (Github Actions). There are two ways we mainly use to trigger workflows, manually and via pushing Github tags. Manual triggering would be the best since it supports non-technical users as well. If we can have the APK variant configuration as user friendly as possible we can lean towards leveraging the manual workflow. More on [manual Github workflows here](https://docs.github.com/en/actions/using-workflows/manually-running-a-workflow).

**Objectives** 
- Store and retrieve APK configurations, including secrets via Vault
- For each flavor and environment configuration, generate an APK on release
- Provide a way to download the APKs after the process is complete

The detailed design of the approach is documented here : [Automated APK Release Management](https://docs.google.com/document/d/1neItpNX-gdhUscFbX_rEHSXzG4Km8SsQNet0Bo0LkvU)

### PR DESCRIPTION
**Pull request by ndegwamartin**
**Manual APK release Github Actions workflow configuration**

Closes #2883

### CODE REVIEW

**Review by f-odhiambo — CHANGES_REQUESTED** [review]
Where are the docs to run this ? @ndegwamartin  :)

**Comment by ellykits** [comment]
Has this been tested @f-odhiambo and @ndegwamartin?

**Comment by ndegwamartin** [comment]
@ellykits if you mean under development then yep!

Check the **APK** and **AAB** artifacts here:

- https://github.com/opensrp/fhircore/actions/runs/8099506488
- https://github.com/opensrp/fhircore/actions/runs/8109145225

File: android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt [hunk 1]
```diff
@@ -255,9 +255,12 @@ class PatientExtensionTest : RobolectricTest() {
...
 
   @Test
-  fun testExtractAgeShouldReturnCallGetAgeStringFromDaysWhenPatientHasBirthDate() {
-    val calendar =
-      Calendar.getInstance().apply { timeInMillis = (timeInMillis - (1L * 365 * 24 * 3600 * 1000)) }
+  fun testExtractAgeShouldReturnAgeStringFromDaysWhenPatientHasBirthDate() {
+    val calendar = Calendar.getInstance()
+    calendar.apply {
+      timeInMillis =
+        (timeInMillis - (1L * calendar.getActualMaximum(Calendar.DAY_OF_YEAR) * 24 * 3600 * 1000))
```
Comment (pld) [inline android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt:262]: I think this fix from @marklosh looks cleaner to me, fewer special numbers :D, https://github.com/opensrp/fhircore/pull/3116/files#diff-f7b77a5722cb99a28fd93c5bab900a5b20dfc103e66d798be9ccb8194c5c64f4R265

Comment (ndegwamartin) [reply android/engine/src/test/java/org/smartregister/fhircore/engine/util/extension/PatientExtensionTest.kt:262]: Agreed, reverting this

**Review by f-odhiambo — APPROVED** [review]
LGTM

**Pull request merged**